### ISSUE TITLE
[ap,android] proper response to methods when using LocalOnlyHostspot

### ISSUE THREAD

**Issue by daadu** [alternadom/WiFiFlutter#134]
Methods that needs to be re-worked to send proper response when using `LocalOnlyHostspot`:
- `getWiFiAPSSID()` - get "random SSID" generated for `LocalOnlyHostspot`
- `getWiFiAPPreSharedKey()` - get "random pre shared key" generated for `LocalOnlyHostspot`
- `isWiFiAPEnabled()` - check if `LocalOnlyHostspot` hotspot is created

**Comment by daadu** [comment]
`isSSIDHidden` - should return `true`, if `LocalOnlyHostspot`  is on.

**Comment by fedepell** [comment]
@daadu : I believe it should rather return what `isHiddenSsid()` of `SoftApConfiguration` that is returned to the `onStarted` calback. Because I'm not sure it is really _hidden_ as in "doesn't broadcast its SSID" or that it may change from version to version or so.

**Comment by daadu** [comment]
@fedepell  can we request to have "hidden Hostpost" with  `LocalOnlyHostspot`?

**Comment by fedepell** [comment]
@daadu : no, with `LocalOnlyHotspot` as far as I can understand you cannot ask any "details", including SSID being hidden or not. So it's all up to what the OS will generate. Which I don't know how much it may depend on the version (ie. what is valid for 10 may not be valid in the future) or specific vendor. 

That's why, in my opinion, using the info from the specific object returned, although slightly more complicated to code, would guarantee a better functionality and longer lifespan of the code.

**Comment by mavyfaby** [comment]
`SoftApConfiguration` doesn't support Android SDK 29 as it is introduced in SDK 30.

**Comment by daadu** [comment]
> That's why, in my opinion, using the info from the specific object returned, although slightly more complicated to code, would guarantee a better functionality and longer lifespan of the code.

@fedepell Can you explain this, I did not get WDYM by " using the info from the specific object returned"?

**Comment by daadu** [comment]
`LocalOnlyHotspot` is since API 26, and we use it only for >= 29. I assume because the hidden API used for AP works till then, but should enable it for 26+ (instead of current 29+), since it is the official API. If not then we should have flag argument like `localOnly: false` that enables it for 26+.

**Comment by fedepell** [comment]
@daadu : what I meant is that we could ask the info from `SoftApConfiguration` (see [link](https://developer.android.com/reference/android/net/wifi/SoftApConfiguration)). But as @mavyfaby correctly mentioned, that is available from 30 up, so we could make it work just for `>=30`

Sad to say but with all these continuous changes in the API support, it is quite hard to be effective and have decently clean code :(

**Comment by dirkvranckaert** [comment]
Is there any progress on this? When can we expect a release with this feature?

**Comment by daadu** [comment]
@dirkvranckaert It is being implemented at #268, do test your app against that PR and give your feedback there.

**Comment by daadu** [comment]
This feature has landed on pub.dev with [v0.3.16](https://pub.dev/packages/wifi_iot/versions/0.3.16).

@PoloLacoste Thanks for your contribution, check yourself in [CONTRIBUTORS.md](https://github.com/flutternetwork/WiFiFlutter/blob/master/CONTRIBUTORS.md)!
